assets/armsnippets.jsonc

```diff
@@ -140,6 +140,120 @@
         ],
         "description": "Application Service Plan (Server Farm)"
     },
+    "Application Gateway": {
+        "prefix": "arm-app-gw",
+        "body": [
+            "{",
+            "    \"name\": \"${1:applicationGatewayName}\",",
+            "    \"type\": \"Microsoft.Network/applicationGateways\",",
+            "    \"apiVersion\": \"2019-11-01\",",
+            "    \"location\": \"[resourceGroup().location]\",",
+            "    \"dependsOn\": [",
+            "        \"[resourceId('Microsoft.Network/virtualNetworks', '${2:vnetName}')]\",",
+            "        \"[resourceId('Microsoft.Network/publicIPAddresses', '${3:publicIP}')]\"",

```

Comment:
Change name to `publicIPAddress1` to allign with PIP snippet.

![image](https://user-images.githubusercontent.com/7844635/77959051-d9010f00-728a-11ea-9488-7867eaf59860.png)